### PR TITLE
Sorr task283 create test list to run with sorrentum 2

### PR DESCRIPTION
**Pull request by LibertasSpZ**
Progress in #500 :

- Sorry for the multi-commits. The commits were used to track different types of failures, like `timeout > 5s` v.s. `requires_aws`. Change in `helpers/lib_tasks_pytest.py` was made in the last commit.

- Disabled all previously existing failures and ran `i run_fast_tests` again. Most failures are gone. Some new failures emerged. Full output here [output_Aug2.txt](https://github.com/sorrentum/sorrentum/files/12248381/output_Aug2.txt).  Failures start from Line 2823.

Things to notice: 

[1] We have 23 failures now. Many of'em point to `/app/helpers/hmoto.py::setUp`, but I already marked that function as `requires_ck_infra` before running the test. Not sure why it wasn't skipped. 

[2] new types of failures emerged, such as:
    
         ```
         RuntimeError:
         --------------------------------------------------------------------------------
         ACTUAL vs EXPECTED: Test_build_run_command_line1.test_run_fast_tests1
         ```

Not sure what "ACTUAL vs EXPECTED" refers to.
[3] As someone previously mentioned in #283 , some tests were run twice, once passed and once error. e.g.

        ```
        market_data/test/test_real_time_market_data.py::TestRealTimeMarketData2::test_get_twap_price1 PASSED                    
        [ 35%]
        market_data/test/test_real_time_market_data.py::TestRealTimeMarketData2::test_get_twap_price1 ERROR                     
        [ 35%]
        ```

Thank you.

### CODE REVIEW

File: helpers/lib_tasks_pytest.py [hunk 1]
```diff
@@ -152,8 +162,16 @@ def _build_run_command_line(
...
+    is_laptop = not hserver.is_dev_ck()
+    if is_laptop:
+        # Make the timeout much larger.
+        pytest_opts_tmp.append("-o timeout_func_only=true")
+        timeout_in_sec *= 10
+        pytest_opts_tmp.append(f"--timeout {timeout_in_sec}")
+    else:
+        # Add a timeout.
+        pytest_opts_tmp.append("-o timeout_func_only=true")
+        pytest_opts_tmp.append(f"--timeout {timeout_in_sec}")
```
Comment (DanilYachmenev) [inline helpers/lib_tasks_pytest.py:174]: also what about this one? part of what task is it?

File: helpers/test/outcomes/TestCheckDataFrame1.test_check_df_not_equal3/output/test_df.txt [hunk 2]
```diff
@@ -1 +1 @@
-abc
+abc
```
Comment (DanilYachmenev) [inline helpers/test/outcomes/TestCheckDataFrame1.test_check_df_not_equal3/output/test_df.txt:1]: after you make all the changes pls remove these outcome files from the committed changes

File: im/kibot/metadata/test/test_kibot_metadata.py [hunk 3]
```diff
@@ -9,6 +11,9 @@
 
 
 class TestKibotMetadata(hunitest.TestCase):
+    
+    @pytest.mark.requires_aws 
+    @pytest.mark.requires_ck_infra
```
Comment (DanilYachmenev) [inline im/kibot/metadata/test/test_kibot_metadata.py:16]: do we need double markers?
wouldn't `requires_ck_infra` be enough?

File: im_v2/common/data/client/test/test_historical_pq_clients.py [hunk 4]
```diff
@@ -19,6 +19,8 @@
 
 
 class TestHistoricalPqByTileClient1(icdc.ImClientTestCase):
+
```
Comment (DanilYachmenev) [inline im_v2/common/data/client/test/test_historical_pq_clients.py:22]: extra empty line

File: im_v2/ig/data/client/test/test_ig_historical_pq_by_date_taq_bar_client.py [hunk 5]
```diff
@@ -1,4 +1,6 @@
 import logging
+import pytest
+
```
Comment (DanilYachmenev) [inline im_v2/ig/data/client/test/test_ig_historical_pq_by_date_taq_bar_client.py:3]: extra empty line

File: pytest.ini [hunk 6]
```diff
@@ -27,6 +27,10 @@ addopts =
   --durations=3
 
 markers=
+  requires_ck_infra: tests that require to run inside CK infrastructure. More specifically
+    when running Sorrentum developers run on a machine outside the infra, so we need to
```
Comment (DanilYachmenev) [inline pytest.ini:31]: not completed

File: helpers/test/outcomes/TestCheckDataFrame1.test_check_df_not_equal3/output/test_df.txt (hunk 2 above)
Comment (gpsaggese) [reply helpers/test/outcomes/TestCheckDataFrame1.test_check_df_not_equal3/output/test_df.txt:1]: If you do, you can pick up master
```
> i git_merge_master
```
Of course, only on a Git client without changes

File: helpers/lib_tasks_pytest.py [hunk 7]
```diff
@@ -138,10 +138,20 @@ def _build_run_command_line(
...
+    skip_ck_infra_tests = (not hserver.is_dev_ck() or hgit.get_repo_name() ==
+                     "Sorrentum")
+    if skip_ck_infra_tests:
+        # markers = "-m 'not requires_ck_infra'"
+        skipped_tests += " and not requires_ck_infra"
+        # Disable the timeout by removing this part
+        # -o timeout_func_only=true \
+        # -timeout 5 \
+        #
+
```
[the remark this answers is not in the document]
Comment (gpsaggese) [reply helpers/lib_tasks_pytest.py:154]: Yes, that's the right approach.

Since the logic of this function is pretty complicated, we have unit tests `Test_build_run_command_line1`

You want to add some unit tests using `mock` to inject certain values of `hserver.is_dev_ck()` so you can make sure that the right command line is generated.

File: helpers/lib_tasks_pytest.py [hunk 8]
```diff
@@ -138,10 +138,20 @@ def _build_run_command_line(
...
     if custom_marker != "":
```
Comment (gpsaggese) [inline helpers/lib_tasks_pytest.py:155]: It might be too difficult to build the string directly so you could keep `markers` as list of strings and then merge with

```
markers_as_str = " and ".join(["-m {marker}" for market in markers])
```

In other words separate abstract the representation

Makes sense?

File: helpers/lib_tasks_pytest.py [hunk 9]
```diff
@@ -152,8 +162,16 @@ def _build_run_command_line(
     # Adding `timeout_func_only` is a workaround for
     # https://github.com/pytest-dev/pytest-rerunfailures/issues/99. Because of
     # it, we limit only run time, without setup and teardown time.
-    pytest_opts_tmp.append("-o timeout_func_only=true")
-    pytest_opts_tmp.append(f"--timeout {timeout_in_sec}")
+    is_laptop = not hserver.is_dev_ck()
+    if is_laptop:
+        # Make the timeout much larger.
+        pytest_opts_tmp.append("-o timeout_func_only=true")
+        timeout_in_sec *= 10
```
Comment (gpsaggese) [inline helpers/lib_tasks_pytest.py:169]: You can probably just leave the timeout_in_sec *= 10 in the if branch and converge the code paths

File: helpers/hmoto.py [hunk 10]
```diff
@@ -40,6 +40,8 @@ class S3Mock_TestCase(hunitest.TestCase):
     #   Used in some tests that are obtaining data from 3rd party providers.
     binance_secret = None
 
+    @pytest.mark.requires_aws
```
Comment (gpsaggese) [inline helpers/hmoto.py:43]: Let's just add `requires_ck_infra` to disable the broken tests and get regressions back to sanity.

Then we will go back and decide which tests requires what resource.
Finally based on where you are running (laptop, ck server, ...) we map the resources missing or available and the tests will be skipped or not automatically.

Do you see the architecture of this?

File: core/signal_processing/special_functions.py [hunk 11]
```diff
@@ -14,7 +14,7 @@
 _LOG = logging.getLogger(__name__)
 
 
-def calculate_tau_from_com(com: float) -> Union[float, np.float]:
+def calculate_tau_from_com(com: float) -> Union[float, np.float64]:
```
Comment (gpsaggese) [inline core/signal_processing/special_functions.py:17]: ok but each PR should be "independent". One PR to fix float64, one for `@pytest.mark.requires_ck_infra`

We have lots of invoke workflows to help you with managing Git feature branches.

**Review by gpsaggese — COMMENTED** [review]
Going in the right direction.

Let's focus on small PR. The first goal is disable the broken tests, nothing else

**Comment by gpsaggese** [comment]
Good progress. This is a tricky issue: for this reason we need to do small PRs and small iterations.
We can do another PP, if you need help. 

1) For moto, you should skip the entire class and not `setUp`

class S3Mock_TestCase(hunitest.TestCase):

setUp is special method for pytest (no worries you'll learn about this stuff once you start using it more)

2) `new types of failures emerged, such as:`
- The problem IMO is that either you are not caught up with changes in master or you need to revert some changes to the golden `helpers/test/outcomes/TestCheckString1.test_check_string1/output/test.txt`
Let's ignore for now and we can do a mini-PP to fix together

3) Some classes need to be completely disabled, e.g., `TestRealTimeMarketData2` since they do some work in the setUp / tearDown

File: helpers/lib_tasks_pytest.py (hunk 1 above)
Comment (LibertasSpZ) [reply helpers/lib_tasks_pytest.py:174]: This was mostly from my last meeting with @gpsaggese . From what I understand, if the system detects it's not run on the CK infrastructure, then it makes the timeout for fast tests 10 times longer to kill the `timeout: >5 s` issue.

File: helpers/test/outcomes/TestCheckDataFrame1.test_check_df_not_equal3/output/test_df.txt (hunk 2 above)
Comment (LibertasSpZ) [reply helpers/test/outcomes/TestCheckDataFrame1.test_check_df_not_equal3/output/test_df.txt:1]: Yeah this has always been weird: when I `git add` I did not add these files to be committed [I believe each time I only added what I really changed], but for some reason they appear here anyway.

File: im/kibot/metadata/test/test_kibot_metadata.py (hunk 3 above)
Comment (LibertasSpZ) [reply im/kibot/metadata/test/test_kibot_metadata.py:16]: For #500 disabling failures yes. For #501 it looks like we want to dig into reasons for failure so labeling the `requires_aws` now might save some time later.

File: core/signal_processing/special_functions.py (hunk 11 above)
Comment (LibertasSpZ) [reply core/signal_processing/special_functions.py:17]: But if I do not fix the `np.float` issue, `i run_fast_test` exits with ERROR 4 or 2 and no test is actually run.

File: helpers/lib_tasks_pytest.py (hunk 8 above)
Comment (LibertasSpZ) [reply helpers/lib_tasks_pytest.py:155]: I think the markers are handled by calling `_select_tests_to_skip` in line 143 and then later calling `pytest_opts_temp.append` in lines 156, 158 etc. Do we still need to do more on markers?

File: helpers/lib_tasks_pytest.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (LibertasSpZ) [reply helpers/lib_tasks_pytest.py:154]: I don't quite get why `hgit.get_repo_name() == "Sorrentum"` indicates it's a laptop. 

Shouldn't it be the other way around, i.e. `(not hserver.is_ck_dev()) or (hgit.get_repo_name() != "Sorrentum")` indicates laptop? my local directory is named `sorrentum1`, for example.

[the remark this answers is not in the document]
Comment (gpsaggese) [reply helpers/lib_tasks_pytest.py:154]: It's a bit tricky to explain. Let's assume for now
```
is_laptop = not hserver.is_ck_dev()
```

File: helpers/lib_tasks_pytest.py (hunk 1 above)
Comment (gpsaggese) [reply helpers/lib_tasks_pytest.py:174]: Correct.

File: helpers/test/outcomes/TestCheckDataFrame1.test_check_df_not_equal3/output/test_df.txt (hunk 2 above)
Comment (gpsaggese) [reply helpers/test/outcomes/TestCheckDataFrame1.test_check_df_not_equal3/output/test_df.txt:1]: No worries. You'll become more expert about git and its corner cases.

File: im/kibot/metadata/test/test_kibot_metadata.py (hunk 3 above)
Comment (gpsaggese) [reply im/kibot/metadata/test/test_kibot_metadata.py:16]: Ok if you want leave `requires_aws` for now, but let's focus on the simplest possible PR for now so we can merge.

File: core/signal_processing/special_functions.py (hunk 11 above)
Comment (gpsaggese) [reply core/signal_processing/special_functions.py:17]: ok to keep it, but it's just weird that you have this problem running inside the container.

File: helpers/lib_tasks_pytest.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (LibertasSpZ) [reply helpers/lib_tasks_pytest.py:154]: Cool. Thanks!

File: helpers/lib_tasks_pytest.py (hunk 9 above)
Comment (LibertasSpZ) [reply helpers/lib_tasks_pytest.py:168]: done.

File: helpers/hmoto.py (hunk 10 above)
Comment (LibertasSpZ) [reply helpers/hmoto.py]: kind of saw the architecture, but will hopefully learn more. By discussions above we may want to keep the `requires_aws` for a while.

File: helpers/lib_tasks_pytest.py [hunk 12]
```diff
@@ -138,10 +138,20 @@ def _build_run_command_line(
...
+    # Detect if we are running on a laptop.
+    skip_ck_infra_tests = (not hserver.is_dev_ck() or hgit.get_repo_name() ==
+                     "Sorrentum")
+    if skip_ck_infra_tests:
+        # markers = "-m 'not requires_ck_infra'"
+        skipped_tests += " and not requires_ck_infra"
+        # Disable the timeout by removing this part
+        # -o timeout_func_only=true \
+        # -timeout 5 \
+        #
```
Comment (gpsaggese) [inline helpers/lib_tasks_pytest.py:153]: Let's remove this comment

File: helpers/lib_tasks_pytest.py [hunk 13]
```diff
@@ -138,10 +138,20 @@ def _build_run_command_line(
...
     pytest_opts_tmp = []
-
     # Select tests to skip based on the `test_list_name` (e.g., fast tests)
     # and on the custom marker, if present.
     skipped_tests = _select_tests_to_skip(test_list_name)
+    # Detect if we are running on a laptop.
+    skip_ck_infra_tests = (not hserver.is_dev_ck() or hgit.get_repo_name() ==
+                     "Sorrentum")
+    if skip_ck_infra_tests:
+        # markers = "-m 'not requires_ck_infra'"
```
Comment (gpsaggese) [inline helpers/lib_tasks_pytest.py:148]: Let's remove this comment

File: helpers/lib_tasks_pytest.py [hunk 14]
```diff
@@ -138,10 +138,20 @@ def _build_run_command_line(
     )
     pytest_opts = pytest_opts or "."
     pytest_opts_tmp = []
-
     # Select tests to skip based on the `test_list_name` (e.g., fast tests)
     # and on the custom marker, if present.
     skipped_tests = _select_tests_to_skip(test_list_name)
+    # Detect if we are running on a laptop.
+    skip_ck_infra_tests = (not hserver.is_dev_ck() or hgit.get_repo_name() ==
```
Comment (gpsaggese) [inline helpers/lib_tasks_pytest.py:145]: Ok let's just do
```
skip_ck_infra_tests = not hserver.is_dev_ck()
if skip_ck_infra_tests:
    _LOG.warning("Skipping CK infra related tasks")
```

File: helpers/lib_tasks_pytest.py [hunk 15]
```diff
@@ -152,6 +162,11 @@ def _build_run_command_line(
     # Adding `timeout_func_only` is a workaround for
     # https://github.com/pytest-dev/pytest-rerunfailures/issues/99. Because of
     # it, we limit only run time, without setup and teardown time.
+    is_laptop = not hserver.is_dev_ck()
+    if is_laptop:
+        # Make the timeout much larger.
+        timeout_in_sec *= 10
+        # Add a timeout.
```
Comment (gpsaggese) [inline helpers/lib_tasks_pytest.py:169]: Let's remove this

File: helpers/test/test_git.py [hunk 16]
```diff
@@ -212,10 +212,11 @@ def test_get_task_prefix_from_repo_short_name1(self) -> None:
         exp = "DevToolsTask"
         self.assert_equal(act, exp)
 
-
+# Test_git_path1 constantly hangs when I run `i run_fast_tests`.
```
Comment (gpsaggese) [inline helpers/test/test_git.py:215]: -> TODO(shaopengz): This test hangs in `hgit.is_in_amp_as_supermodule()`.

File: helpers/test/test_git.py [hunk 17]
```diff
@@ -212,10 +212,11 @@ def test_get_task_prefix_from_repo_short_name1(self) -> None:
...
+@pytest.mark.requires_ck_infra
 class Test_git_path1(hunitest.TestCase):
     @pytest.mark.skipif(
-        not hgit.is_in_amp_as_supermodule(),
+        not hgit.is_in_amp_as_supermodule(), #error msg says it hangs here.
```
Comment (gpsaggese) [inline helpers/test/test_git.py:219]: You can remove this

File: helpers/test/test_lib_tasks_pytest.py [hunk 18]
```diff
@@ -42,13 +43,32 @@ def test_run_fast_tests1(self) -> None:
             tee_to_file,
             n_threads,
         )
-        exp = (
-            'pytest -m "not slow and not superslow" . '
-            "-o timeout_func_only=true --timeout 5 --reruns 2 "
-            '--only-rerun "Failed: Timeout" -n 1'
-        )
+        skip_ck_infra_tests = (not hserver.is_dev_ck() or hgit.get_repo_name() ==
```
Comment (gpsaggese) [inline helpers/test/test_lib_tasks_pytest.py:46]: Ok for now but we'll do mocking to isolate the server deps.

File: helpers/test/test_lib_tasks_pytest.py [hunk 19]
```diff
@@ -71,6 +91,8 @@ def test_run_fast_tests2(self) -> None:
             tee_to_file,
             n_threads,
         )
+        #TODO(SP): for exp, there has to be similar changes like run_fast_tests1 according to
```
Comment (gpsaggese) [inline helpers/test/test_lib_tasks_pytest.py:94]: -> TODO(shaopengz): Apply the same changes as in `run_fast_tests1`. Disabling it for now.

**Review by gpsaggese — COMMENTED** [review]
Cosmetic changes and then we can merge.
Good job

**Review by gpsaggese — COMMENTED** [review]
Good job
